### PR TITLE
Feature oauth new model

### PR DESCRIPTION
**Pull request by Hamaad102**
### What this PR does (required):
Corrected issue with model not presenting the necessary information. I also separated the code into the correct files (functions that are meant from controller were move from route into auth controller for example). Also contains preliminary files for the scheduler on the front-end which are incomplete at this time.

### Screenshots / Videos (required):
Screenshot (15)

### Any information needed to test this feature (required):
The signup works as normal it just holds more data in the backend than before.

### CODE REVIEW

File: client/src/App.tsx [hunk 1]
```diff
@@ -1,31 +1,44 @@
...
-              <Route exact path="/signup" component={Signup} />
-              <Route exact path="/google" component={GoogleConnect} />
-              <Route exact path="/dashboard">
-                <Dashboard />
-              </Route>
-              <Route path="*">
-                <Redirect to="/login" />
-              </Route>
-            </Switch>
+            <SocketProvider>
```
Comment (moffatethan) [inline client/src/App.tsx]: I don't see using Socket just yet in this project, the provider could be removed.

File: client/src/interface/User.ts [hunk 2]
```diff
@@ -1,6 +1,10 @@
 export interface User {
-  email: string;
   username: string;
+  email: string;
+  strategy: string;
+  id: number;
+  picture: string;
+  mongoId: number;
```
Comment (moffatethan) [inline client/src/interface/User.ts:7]: Could you explain this here? MongoIDs aren't numbers and there is already an id which is also a number too?

Comment (moffatethan) [inline client/src/interface/User.ts]: You'll want to update this in the future to be a string because MongoIDs are strings. Also, where do we have two ids here?

File: server/app.js [hunk 3]
```diff
@@ -38,28 +42,34 @@ io.on("connection", (socket) => {
...
+// Middleware
 app.use(json());
 app.use(urlencoded({ extended: false }));
 app.use(cookieParser());
 app.use(express.static(join(__dirname, "public")));
 
+app.use((req, res, next) => {
+  req.io = io;
+  next();
+});
```
Comment (moffatethan) [inline server/app.js]: We won't need Socket for this project, this can be removed in another PR.

**Review by moffatethan — COMMENTED** [review]
Left some comments for you, otherwise it's looking good. Not sure why `sample.env` was deleted. That's something we want to keep if future collaborators ever joined they would know what environment variables are needed to run the project.

File: client/src/interface/User.ts (hunk 2 above)
Comment (Hamaad102) [reply client/src/interface/User.ts]: Id represents the value associated with the auth service (google and apple for the future). I kept them both in case we needed the mongo value to quickly access the data in database though I can remove that.

**Review by moffatethan — APPROVED** [review]
Good to go! Thanks for doing the changes
